tracing/tempo/http_client.go

```diff
@@ -26,9 +27,26 @@ type OtelHTTPClient struct {
 	ClusterTag bool
 }
 
+var (
+	once       sync.Once
+	tempoCache TempoCache
+)
```

Comment:
Can we avoid global variables and instead pass the cache into the client? This would make it easier to test as well and will avoid side effects for anything that is testing this package.